conformance/02_push_test.go

```diff
@@ -60,7 +60,7 @@ var test02Push = func() {
 				Expect(resp.StatusCode()).To(Equal(http.StatusNotFound))
 			})
 
-			g.Specify("POST request with digest and blob should yield a 201", func() {
+			g.Specify("POST request with digest and blob should yield a 201 or 202", func() {
```

Comment:
This is a bit too lenient. It's not that either 201 or 202 is acceptable, but that 201 and 202 indicate completely different things.

Reply:
From a testing POV, how do you think we should address? Use an env var `UPLOAD_EXPECTED_STATUS=202`? As far as I understand they indicate different things from registry standpoint, but how can this be tested better in the client here?

Reply:
So for cross-repo mounting, 201 indicates that the mount succeeded, and 202 indicates that you should proceed with the upload at the returned location.

I would expect that if a registry returned a 202 to me when I tried to do a monolithic upload, I need to perform a non-monolithic upload at `Location`, instead.

For this particular test, perhaps you don't care about the distinction? Quite possibly either response is valid, but I would guess that the rest of the test would break if a 202 was returned, given that the config wasn't uploaded, but I haven't looked through this code too deeply. If the upload _did_ succeed, I would always expect a 201.

Is there a registry returning a 202 here? I would take that to indicate "I don't support single-request monolithic uploads", which maybe should fail the test? I'd need a bit more context to understand, I think.

Reply:
My guess is this is for Harbor, which uses docker/distribution as a backend. So I believe that to be the context. Docker distribution is returning a 202 for this test, causing Harbor (and docker distribution) to fail.

So do we fix the test (as done here) to be more lenient, or is 202 actually incorrect as it is encoded in docker/distribution? Personally I lean towards the former since lotsa registries seem to have roots in docker

Reply:
Reading through [some code](https://github.com/distribution/distribution/blob/a637ebcd0164e27be6ee50a5df95655e51b93e18/registry/handlers/blobupload.go#L61-L101) there doesn't _appear_ to be anything that handles single-request monolithic blob uploads (@dmcgowan might be able to confirm). I don't see anything that reads the body, so I am assuming it's this case:

>  I would take that to indicate "I don't support single-request monolithic uploads", which maybe should fail the test?

I'm guessing the other upload paths are succeeding to upload the config, so this doesn't affect the other tests, even though the upload fails.

Reply:
I would be happy if we added spec language to that effect. It seems like a reasonable failure mode.

We already have that for cross-repo mounts, but it should apply to these monolithic uploads as well.